diff --git a/src/Utils.cpp b/src/Utils.cpp
index 13073c3..54ef5f8 100644
--- a/src/Utils.cpp
+++ b/src/Utils.cpp
@@ -702,6 +702,17 @@ bool Utils::IsRPHotkeySet() {
 	return (count > 0);
 }
 
+const char* Utils::GetLastReplayBufferFilePath()
+{
+	OBSOutputAutoRelease output = obs_frontend_get_replay_buffer_output();
+	calldata_t cd = { 0 };
+	proc_handler_t* ph = obs_output_get_proc_handler(output);
+	proc_handler_call(ph, "get_last_replay", &cd);
+	auto ret = calldata_string(&cd, "path");
+	calldata_free(&cd);
+	return ret;
+}
+
 const char* Utils::GetFilenameFormatting() {
 	config_t* profile = obs_frontend_get_profile_config();
 	return config_get_string(profile, "Output", "FilenameFormatting");
diff --git a/src/Utils.h b/src/Utils.h
index 85c5c86..3069542 100644
--- a/src/Utils.h
+++ b/src/Utils.h
@@ -78,6 +78,7 @@ namespace Utils {
 	QString ParseDataToQueryString(obs_data_t* data);
 	obs_hotkey_t* FindHotkeyByName(QString name);
 
+	const char* GetLastReplayBufferFilePath();
 	bool ReplayBufferEnabled();
 	void StartReplayBuffer();
 	bool IsRPHotkeySet();
diff --git a/src/WSEvents.cpp b/src/WSEvents.cpp
index d4b18e7..4fa9c17 100644
--- a/src/WSEvents.cpp
+++ b/src/WSEvents.cpp
@@ -225,6 +225,10 @@ void WSEvents::FrontendEventHandler(enum obs_frontend_event event, void* private
 		case OBS_FRONTEND_EVENT_REPLAY_BUFFER_STOPPED:
 			owner->OnReplayStopped();
 			break;
+			
+		case OBS_FRONTEND_EVENT_REPLAY_BUFFER_SAVED:
+			owner->OnReplaySaved();
+			break;
 
 		case OBS_FRONTEND_EVENT_STUDIO_MODE_ENABLED:
 			owner->OnStudioModeSwitched(true);
@@ -858,6 +862,20 @@ void WSEvents::OnReplayStopped() {
 	broadcastUpdate("ReplayStopped");
 }
 
+/**
+* Replay Buffer saved successfully
+*
+* @api events
+* @name ReplaySaved
+* @category replay buffer
+* @since 4.2.0
+*/
+void WSEvents::OnReplaySaved() {
+	OBSDataAutoRelease data = obs_data_create();
+	obs_data_set_string(data, "replayFilename", Utils::GetLastReplayBufferFilePath());
+	broadcastUpdate("ReplaySaved", data);
+}
+
 /**
  * OBS is exiting.
  *
diff --git a/src/WSEvents.h b/src/WSEvents.h
index f0c4cf6..ed30872 100644
--- a/src/WSEvents.h
+++ b/src/WSEvents.h
@@ -111,6 +111,7 @@ private:
 	void OnReplayStarted();
 	void OnReplayStopping();
 	void OnReplayStopped();
+	void OnReplaySaved();
 
 	void OnStudioModeSwitched(bool enabled);
 	void OnPreviewSceneChanged();
