diff --git a/obs-websocket-dotnet/IOBSWebsocket.cs b/obs-websocket-dotnet/IOBSWebsocket.cs
index 7913cbb..e602e76 100644
--- a/obs-websocket-dotnet/IOBSWebsocket.cs
+++ b/obs-websocket-dotnet/IOBSWebsocket.cs
@@ -5,6 +5,7 @@ using OBSWebsocketDotNet.Communication;
 using OBSWebsocketDotNet.Types;
 using OBSWebsocketDotNet.Types.Events;
 using Websocket.Client;
+using System.Threading.Tasks;
 
 namespace OBSWebsocketDotNet
 {
@@ -1050,7 +1051,7 @@ namespace OBSWebsocketDotNet
         /// </summary>
         /// <param name="url">Server URL in standard URL format.</param>
         /// <param name="password">Server password</param>
-        void ConnectAsync(string url, string password);
+        Task ConnectAsync(string url, string password);
 
         /// <summary>
         /// Disconnect this instance from the server
diff --git a/obs-websocket-dotnet/OBSWebsocket.cs b/obs-websocket-dotnet/OBSWebsocket.cs
index b3c3770..c4b4987 100644
--- a/obs-websocket-dotnet/OBSWebsocket.cs
+++ b/obs-websocket-dotnet/OBSWebsocket.cs
@@ -85,7 +85,9 @@ namespace OBSWebsocketDotNet
         /// </summary>
         /// <param name="url">Server URL in standard URL format.</param>
         /// <param name="password">Server password</param>
-        public void ConnectAsync(string url, string password)
+        /// <returns>Returns the awaitable Task from <see cref="WebsocketClient.StartOrFail"/>.
+        /// NOTE: After awaiting, the client is still not ready to work. Please subscribe to the Connected event to determine when the connection is actually fully established</returns>
+        public Task ConnectAsync(string url, string password)
         {
             if (!url.ToLower().StartsWith(WEBSOCKET_URL_PREFIX))
             {
@@ -105,7 +107,7 @@ namespace OBSWebsocketDotNet
             wsConnection.DisconnectionHappened.Subscribe(d => Task.Run(() => OnWebsocketDisconnect(this, d)));
 
             connectionPassword = password;
-            wsConnection.StartOrFail();
+            return wsConnection.StartOrFail();
         }
 
         /// <summary>
