diff --git a/obs-websocket-dotnet/Events.cs b/obs-websocket-dotnet/Events.cs
index 4a3504f..43c07fc 100644
--- a/obs-websocket-dotnet/Events.cs
+++ b/obs-websocket-dotnet/Events.cs
@@ -84,6 +84,12 @@ namespace OBSWebsocketDotNet
         /// </summary>
         public event EventHandler<SceneTransitionVideoEndedEventArgs> SceneTransitionVideoEnded;
 
+        /// <summary>
+        /// Triggered for the screenshot feature available in Settings -> Hotkeys -> Screenshot Output ONLY.
+        /// Applications using Get/SaveSourceScreenshot should implement a CustomEvent if this kind of inter-client communication is desired.
+        /// </summary>
+        public event EventHandler<ScreenshotSavedEventArgs> ScreenshotSaved;
+
         /// <summary>
         /// Triggered when switching to another profile
         /// </summary>
@@ -361,6 +367,10 @@ namespace OBSWebsocketDotNet
                     SceneTransitionVideoEnded?.Invoke(this, new SceneTransitionVideoEndedEventArgs((string)body["transitionName"]));
                     break;
 
+                case nameof(ScreenshotSaved):
+                    ScreenshotSaved?.Invoke(this, new ScreenshotSavedEventArgs((string)body["savedScreenshotPath"]));
+                    break;
+
                 case nameof(CurrentProfileChanged):
                     CurrentProfileChanged?.Invoke(this, new CurrentProfileChangedEventArgs((string)body["profileName"]));
                     break;
diff --git a/obs-websocket-dotnet/Types/Events/ScreenshotSavedEventArgs.cs b/obs-websocket-dotnet/Types/Events/ScreenshotSavedEventArgs.cs
new file mode 100644
index 0000000..a09c49b
--- /dev/null
+++ b/obs-websocket-dotnet/Types/Events/ScreenshotSavedEventArgs.cs
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace OBSWebsocketDotNet.Types.Events
+{
+    /// <summary>
+    /// Event args for <see cref="OBSWebsocket.ScreenshotSaved"/>
+    /// </summary>
+    public class ScreenshotSavedEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Path of the saved image file
+        /// </summary>
+        public string SavedScreenshotPath { get; }
+
+        /// <summary>
+        /// Default Constructor
+        /// </summary>
+        /// <param name="savedScreenshotPath">Path of the saved image file</param>
+        public ScreenshotSavedEventArgs(string savedScreenshotPath)
+        {
+            SavedScreenshotPath = savedScreenshotPath;
+        }
+    }
+}
\ No newline at end of file
